--- a/native/Makefile
+++ b/native/Makefile
@@ -23,6 +23,7 @@
 OS=$(shell uname | sed -e 's/\(CYGWIN\|MINGW32\).*/win32/g' \
                         -e 's/SunOS.*/solaris/g' \
                         -e 's/FreeBSD.*/freebsd/g' \
+			-e 's/GNU\/kFreeBSD/linux/g' \
                         -e 's/OpenBSD.*/openbsd/g' \
                         -e 's/Darwin.*/darwin/g' \
                         -e 's/Linux.*/linux/g')
--- a/build.xml
+++ b/build.xml
@@ -126,6 +126,9 @@
     <condition property="os.prefix" value="freebsd-${os.arch}">
       <os name="FreeBSD"/>
     </condition>
+    <condition property="os.prefix" value="linux-${jre.arch}">
+      <os name="GNU/kFreeBSD"/>
+    </condition>
     <condition property="os.prefix" value="openbsd-${os.arch}">
       <os name="OpenBSD"/>
     </condition>
